tardis/apps/publication_workflow/static/publication-form/publication_form.js

```diff
@@ -0,0 +1,504 @@
+        vm.experiments = response.data;
+
+        // Set default experiment
+        for (var i = 0; i < vm.experiments.length; i++) {
+            if (vm.experiments[i].id === vm.experiment) {
+                vm.selectedExperiment = vm.experiments[i];
+                break;
+            }
+        }
+        vm.loadingData = false;
+    });
+
+
+    // Add a dataset to the list of selected datasets
+    vm.addDatasets = function (experiment, datasets) {
+        if (angular.isUndefined(datasets)) {
+            return;
+        }
+        for (var i = 0; i < datasets.length; i++) {
+            vm.formData.addedDatasets.push({
+                "experiment": experiment.title,
+                "experiment_id": experiment.id,
+                "dataset": datasets[i]
+            });
+        }
+    };
+    // Remove a dataset from the list of selected datasets
+    vm.removeDataset = function (dataset) {
+        var index = vm.formData.addedDatasets.indexOf(dataset);
+        if (index > -1) {
+            vm.formData.addedDatasets.splice(index, 1);
+        }
+    };
+
+    // Advance to the next page of the form
+    vm.nextPage = function () {
+        angular.element($document[0].querySelector('.ngdialog')).scrollTop(0);
+        if (vm.currentPageIdx < vm.formPages.length - 1 && !vm.loadingData) {
+            vm.errorMessages = [];
+            vm.infoMessage = "";
+            vm.currentPage.validationFunction(function () { // On success...
+                vm.currentPageIdx++;
+                vm.currentPage = vm.formPages[vm.currentPageIdx];
+            }, function () { // On error...
+                // do nothing...
+            });
+        }
+    };
+    // Move back a page
+    vm.previousPage = function () {
+        if (vm.currentPageIdx - 1 >= 0 && !vm.loadingData) {
+            vm.errorMessages = [];
+            vm.infoMessage = "";
+            vm.currentPageIdx--;
+            vm.currentPage = vm.formPages[vm.currentPageIdx];
+        } else if (vm.currentPageIdx === 0) {
+            ngDialog.close();
+        }
+    };
+
+    vm.isComplete = function () {
+        return vm.currentPageIdx === (vm.formPages.length - 1);
+    };
+
+    vm.isLastPage = function () { // Actually, second last page
+        return vm.currentPageIdx === (vm.formPages.length - 2);
+    };
+
+    // Set the publication title
+    vm.setTitle = function (title) {
+        vm.formData.publicationTitle = title;
+    };
+
+    vm.setDescription = function (description) {
+        vm.formData.publicationDescription = description;
+    };
+
+    // Scroll the dataset list to top
+    vm.scrollDsSelectorToTop = function() {
+        angular.element($document[0].querySelector('#datasetList')).scrollTop(0);
+    };
+
+    // Add author to publication
+    vm.addAuthorEntry = function () {
+        $log.info(vm.formData);
+        vm.formData.authors.push(
+            {
+                'name': '',
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/mytardis/mytardis/pullRequest?prid=7933951)
```suggestion
                "name": '',
```